src/sentry/notifications/platform/service.py

```diff
@@ -218,5 +227,7 @@ def notify_target_async[T: NotificationData](
         # Step 4: Send the notification
         try:
             provider.send(target=target, renderable=renderable)
-        except Exception as e:
-            lifecycle.record_failure(failure_reason=e, create_issue=False)
+        except IntegrationConfigurationError as e:
+            lifecycle.record_halt(halt_reason=e, create_issue=False)
+        except (IntegrationError, Exception) as e:
```

Comment:
It's probably okay to omit `IntegrationError` here as `Exception` should already cover it.